### PR TITLE
feat: 🎸 add useDropdownOptions hook with tests

### PR DESCRIPTION
**Pull request by codyarose**
Based on the `useTransformToOptions` hook from sc2-senior

## Checklist

- [x] Tests for the changes have been added (for bug fixes / features) and all tests are passing
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)

## What is the current behavior?

No `useDropdownOptions` hook

Ref: #8 

## What is the new behavior?

- `useDropdownOptions` hook

### CODE REVIEW

File: src/hooks/useDropdownOptions.test.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import {renderHook} from '@testing-library/react-hooks';
+import {useDropdownOptions} from '.';
+import {Item} from './useDropdownOptions';
```
Comment (SeanGroff) [inline src/hooks/useDropdownOptions.test.ts:3]: I wasn't sure what this import was at first. 

What about:

```suggestion
import type {Item} from './useDropdownOptions';
```

"Back in the day" types were prefaced with the letter I to signify it was an interface or type. I know that's fallen out of favor, what's the general convention now for naming reusable Types?

File: src/hooks/useDropdownOptions.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+
+export interface Item {
```
Comment (SeanGroff) [inline src/hooks/useDropdownOptions.ts:3]: I used to only use interface. Now it seems there aren't many worthwhile benefits to using interface over type. 

When should we use one over the other? Or, which should we use, interface or type?

File: src/hooks/useDropdownOptions.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+ * Our form system expects an Array of Objects with a shape of [{label: String, value: String}]
+ * @param {string} label
+ * @param {string} value
+ * @param {array} items [{item}]
+ */
+export function useDropdownOptions(
+  label: string,
+  value: string,
+  items?: Item[]
+): {label: string | number; value: string | number | null}[] {
```
Comment (SeanGroff) [inline src/hooks/useDropdownOptions.ts]: Inline types can look pretty noisy. Thoughts on defining the type, assigning the type to a variable then using the type variable vs an inline type?

File: src/hooks/useDropdownOptions.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  return React.useMemo(() => {
+    if (items && !isObjectArray(items)) {
+      throw Error(
+        'The items provided to useDropdownOptions must be an array of objects'
+      );
+    }
+
+    if (!items || !items.length) {
```
Comment (SeanGroff) [inline src/hooks/useDropdownOptions.ts]: Assuming TypeScript has ?. You can ditch the ||.  

if (!items?.length) {}

File: src/hooks/useDropdownOptions.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+      return [{label: '- -', value: null}];
+    }
+
+    return items.map((item) => ({label: item[label], value: item[value]}));
+  }, [items, label, value]);
+}
+
+const isObjectArray = (array: Object[]) => {
```
Comment (SeanGroff) [inline src/hooks/useDropdownOptions.ts:34]: I like inline types when they are short and sweet like this one. 

We'll have to put together a TypeScript style guide sometime in the future once we all settle on our preferences.

**Review by SeanGroff — APPROVED** [review]

File: src/hooks/useDropdownOptions.test.ts (hunk 1 above)
Comment (codyarose) [reply src/hooks/useDropdownOptions.test.ts:3]: I don't think I've ever used `import type` before, had to look that one up. 

I'm not sure that there's an agreed upon convention beyond PascalCase now. I try to name them based on their use case, in this instance the arg is called `items`, so `Item` seemed fitting, but that is pretty general. `ItemModel` is a convention I've seen a couple times that's a bit more descriptive

File: src/hooks/useDropdownOptions.ts (hunk 2 above)
Comment (codyarose) [reply src/hooks/useDropdownOptions.ts:3]: That's something I've struggled a bit with, trying to figure out which one is the right choice for each use case. I think there's some good info in [this thread](https://twitter.com/kentcdodds/status/1351662112200822785).

I tend to do the same as what someone else said in that thread: use interfaces when I can and types when I can't.

File: src/hooks/useDropdownOptions.ts (hunk 3 above)
Comment (codyarose) [reply src/hooks/useDropdownOptions.ts]: For sure - but now that I'm looking at it again, the inferred return type is the exact same thing, so I think I could forgo specifying a return type altogether

**Pull request merged**